lib/finder.js

```diff
@@ -7,26 +7,26 @@ const Pnr = require(`./pnr`);
 const FileMockDataReader = require("./fileMockDataReader");
 const WhitelistReader = require("./whitelistReader");
 const Logger = require(`../lib/logger/logger`);
+const Config = require(`./config`);
 const logger = new Logger(`finder.js`);
-
-const defaultPattern = `!(node_modules){,/**}`;
-
 class Finder {
   static findPnrs(searchPath, pattern) {
+    const config = Config.getConfig(searchPath);
     let files = [];
     if (fs.lstatSync(searchPath).isDirectory()) {
-      files = glob.sync(pattern || defaultPattern, {
+      files = glob.sync(pattern || config.pattern, {
         cwd: searchPath,
         absolute: true,
         nodir: true,
-        ignore: this._getIgnoreGlobs(searchPath)
+        ignore: [...this._getIgnoreGlobs(searchPath), ...config.ignorePath]
       });
     } else {
       files.push(searchPath);
     }
     const validator = new Validator([
       ...FileMockDataReader.fetch(),
-      ...WhitelistReader.fetch(searchPath)
+      ...WhitelistReader.fetch(searchPath),
```

Comment:
It might be a good idea to normalize the pnrs we get from WhitelistReader.fetch(searchPath), and config.ignorePnr. If they are in the wrong format in the file they wont work.